pkg/server/config/config.go

```diff
@@ -0,0 +1,326 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"time"
+)
+
+const (
+	DefaultChangelogHorizonOffset           = 0
+	DefaultResolveNodeLimit                 = 25
+	DefaultResolveNodeBreadthLimit          = 100
+	DefaultListObjectsDeadline              = 3 * time.Second
+	DefaultListObjectsMaxResults            = 1000
+	DefaultMaxConcurrentReadsForCheck       = math.MaxUint32
+	DefaultMaxConcurrentReadsForListObjects = math.MaxUint32
+	DefaultCheckQueryCacheLimit             = 10000
+	DefaultCheckQueryCacheTTL               = 10 * time.Second
+	DefaultCheckQueryCacheEnable            = false
+)
+
+// DatastoreConfig defines OpenFGA server configurations for datastore specific settings.
+type DatastoreConfig struct {
+	// Engine is the datastore engine to use (e.g. 'memory', 'postgres', 'mysql')
+	Engine   string
+	URI      string
+	Username string
+	Password string
+
+	// MaxCacheSize is the maximum number of cache keys that the storage cache can store before evicting
+	// old keys. The storage cache is used to cache query results for various static resources
+	// such as type definitions.
+	MaxCacheSize int
+
+	// MaxOpenConns is the maximum number of open connections to the database.
+	MaxOpenConns int
+
+	// MaxIdleConns is the maximum number of connections to the datastore in the idle connection pool.
+	MaxIdleConns int
+
+	// ConnMaxIdleTime is the maximum amount of time a connection to the datastore may be idle.
+	ConnMaxIdleTime time.Duration
+
+	// ConnMaxLifetime is the maximum amount of time a connection to the datastore may be reused.
+	ConnMaxLifetime time.Duration
+}
+
+// GRPCConfig defines OpenFGA server configurations for grpc server specific settings.
+type GRPCConfig struct {
+	Addr string
+	TLS  *TLSConfig
+}
+
+// HTTPConfig defines OpenFGA server configurations for HTTP server specific settings.
+type HTTPConfig struct {
```

Comment:
Thinking out loud here, it's a bit odd to have standalone server configurations in `pkg/server/config`. For example, `GRPCConfig`, `HTTPConfig`, `TLSConfig`, `AuthnConfig`, `TraceConfig`, `ProfilerConfig`, `MetricsConfig`, etc..

These are all configurations that are used to configure features of OpenFGA when it is running as a standalone server as opposed to being imported as a library. All of these are specific to the server as a library:

```
	DefaultChangelogHorizonOffset           = 0
	DefaultResolveNodeLimit                 = 25
	DefaultResolveNodeBreadthLimit          = 100
	DefaultListObjectsDeadline              = 3 * time.Second
	DefaultListObjectsMaxResults            = 1000
	DefaultMaxConcurrentReadsForCheck       = math.MaxUint32
	DefaultMaxConcurrentReadsForListObjects = math.MaxUint32
	DefaultCheckQueryCacheLimit             = 10000
	DefaultCheckQueryCacheTTL               = 10 * time.Second
	DefaultCheckQueryCacheEnable            = false
```

Should we keep standalone server configs internal to OpenFGA since there is no use to import those publicly in another Go app (maybe `internal/server/config`)? What do you think would be the better way to structure these differences?

Reply:
ok, needed to play out what it looked to us openfga as a lib and setup a server. i think moving this over to `internal` is a practical approach. Defaults are already encapsulated when using the available server constructors, all the composing structs for Config are also nicely tucked away into our builtin run command and such